### ISSUE TITLE
[REGRESSION]: UseDotnetV2 fails to extract and set `PATH` sometimes

### ISSUE THREAD

**Issue by sanderaernouts** [microsoft/azure-pipelines-tasks#20535]
### New issue checklist

- [x] I searched for [existing GitHub issues](https://github.com/microsoft/azure-pipelines-tasks/issues)
- [x] I read [pipeline troubleshooting guide](https://docs.microsoft.com/vsts/build-release/actions/troubleshooting)
- [x] I checked [how to collect logs](https://learn.microsoft.com/azure/devops/pipelines/troubleshooting/review-logs?view=azure-devops)

### Task name

UseDotnetV2

### Breaking task version

2.246.1

### Last working task version

2.243.1

### Regression Description

Most of the time, when running the task, I see in the logs that the requested version of .NET is downloaded, extracted, and added to the PATH:
```
Downloading: https://download.visualstudio.microsoft.com/download/pr/12ee34e8-640c-400e-a6dc-4892b442df92/81d40fc98a5bbbfbafa4cc1ab86d6288/dotnet-sdk-6.0.427-linux-x64.tar.gz
Extracting downloaded package /home/vsts/work/_temp/3394240c-8ce2-42cc-a0fe-51e583a86563.
Extracting archive
/usr/bin/tar xC /home/vsts/work/_temp/9cfe3e8b-0101-4594-abad-bea95f2edf43 -f /home/vsts/work/_temp/3394240c-8ce2-42cc-a0fe-51e583a86563

Successfully installed .NET Core sdk version 6.0.427.
Creating global tool path and pre-pending to PATH.
Finishing: use .NET SDK 6.x
```

But sometimes the tasks exits, with a ✅ , before the download is actually completed. So it does not extract and add the .NET version to the PATH:
```
Downloading: https://download.visualstudio.microsoft.com/download/pr/1ebffeb0-f090-4001-9f13-69f112936a70/5dbc249b375cca13ec4d97d48ea93b28/dotnet-sdk-8.0.402-linux-x64.tar.gz
Finishing: use .NET SDK 8.x
```

I use this task to install both .NET 6.x and .NET 8.x and it can randomly fail to actually install one of those.

### Environment type (Please select at least one enviroment where you face this issue)

- [ ] Self-Hosted
- [x] Microsoft Hosted
- [ ] VMSS Pool
- [ ] Container

### Azure DevOps Server type

dev.azure.com (formerly visualstudio.com)

### Operation system

Ubuntu 22.04.5

### Relevant log output

There is no error message, but it looks like the downloading times out and that specific error is not handled by the task, so it returns a success status.
```
Downloading: https://download.visualstudio.microsoft.com/download/pr/1ebffeb0-f090-4001-9f13-69f112936a70/5dbc249b375cca13ec4d97d48ea93b28/dotnet-sdk-8.0.402-linux-x64.tar.gz
Finishing: use .NET SDK 8.x
```

### Full task logs with system.debug enabled

<details>
  <summary>UNSUCCESSFUL RUN</summary>
  <pre>##[debug]Evaluating condition for step: 'use .NET SDK 8.x'
##[debug]Evaluating: SucceededNode()
##[debug]Evaluating SucceededNode:
##[debug]=> True
##[debug]Result: True
Starting: use .NET SDK 8.x
==============================================================================
Task         : Use .NET Core
Description  : Acquires a specific version of the .NET Core SDK from the internet or the local cache and adds it to the PATH. Use this task to change the version of .NET Core used in subsequent tasks. Additionally provides proxy support.
Version      : 2.246.1
Author       : Microsoft Corporation
Help         : https://aka.ms/AA4xgy0
==============================================================================
##[debug]Using node path: /home/vsts/agents/3.245.0/externals/node20_1/bin/node
##[debug]system.debug=true
##[debug]DistributedTask.Tasks.Node.SkipDebugLogsWhenDebugModeOff=True
##[debug]agent.TempDirectory=/home/vsts/work/_temp
##[debug]loading inputs and endpoints
##[debug]loading INPUT_PACKAGETYPE
##[debug]loading INPUT_USEGLOBALJSON
##[debug]loading INPUT_WORKINGDIRECTORY
##[debug]loading INPUT_VERSION
##[debug]loading INPUT_INCLUDEPREVIEWVERSIONS
##[debug]loading INPUT_INSTALLATIONPATH
##[debug]loading INPUT_PERFORMMULTILEVELLOOKUP
##[debug]loading ENDPOINT_AUTH_SYSTEMVSSCONNECTION
##[debug]loading ENDPOINT_AUTH_SCHEME_SYSTEMVSSCONNECTION
##[debug]loading ENDPOINT_AUTH_PARAMETER_SYSTEMVSSCONNECTION_ACCESSTOKEN
##[debug]loading SECRET_SYSTEM_ACCESSTOKEN
##[debug]loaded 11
##[debug]Agent.ProxyUrl=undefined
##[debug]Agent.CAInfo=undefined
##[debug]Agent.ClientCert=undefined
##[debug]Agent.SkipCertValidation=undefined
##[debug]Agent.ProxyUrl=undefined
##[debug]Agent.CAInfo=undefined
##[debug]Agent.ClientCert=undefined
##[debug]check path : /home/vsts/work/_tasks/UseDotNet_b0ce7256-7898-45d3-9cb5-176b752bfea6/2.246.1/node_modules/azure-pipelines-tool-lib/lib.json
##[debug]adding resource file: /home/vsts/work/_tasks/UseDotNet_b0ce7256-7898-45d3-9cb5-176b752bfea6/2.246.1/node_modules/azure-pipelines-tool-lib/lib.json
##[debug]system.culture=en-US
##[debug]Setting resource path to /home/vsts/work/_tasks/UseDotNet_b0ce7256-7898-45d3-9cb5-176b752bfea6/2.246.1/task.json
##[debug]check path : /home/vsts/work/_tasks/UseDotNet_b0ce7256-7898-45d3-9cb5-176b752bfea6/2.246.1/task.json
##[debug]adding resource file: /home/vsts/work/_tasks/UseDotNet_b0ce7256-7898-45d3-9cb5-176b752bfea6/2.246.1/task.json
##[debug]system.culture=en-US
##[debug]check path : /home/vsts/work/_tasks/UseDotNet_b0ce7256-7898-45d3-9cb5-176b752bfea6/2.246.1/node_modules/azure-pipelines-tasks-packaging-common/module.json
##[debug]adding resource file: /home/vsts/work/_tasks/UseDotNet_b0ce7256-7898-45d3-9cb5-176b752bfea6/2.246.1/node_modules/azure-pipelines-tasks-packaging-common/module.json
##[debug]system.culture=en-US
##[debug]useGlobalJson=false
##[debug]packageType=sdk
##[debug]version=8.x
##[debug]vsVersion=undefined
##[debug]nugetVersion=undefined
##[debug]installationPath=/opt/hostedtoolcache/dotnet
##[debug]performMultiLevelLookup=false
##[debug]includePreviewVersions=false
##[debug]workingDirectory=/home/vsts/work/1/s
##[debug]agent.proxyurl=undefined
##[debug]AZURE_HTTP_USER_AGENT=VSTS_8fd5c2b8-726e-4003-977f-90eee064441e_build_286_0
##[debug]testing directory '/opt/hostedtoolcache/dotnet'
##[debug]testing directory '/opt/hostedtoolcache'
##[debug]downloading
##[debug]Content-Length of downloaded file: 212874079
##[debug]creating stream
Finishing: use .NET SDK 8.x
</pre>
</details>

<details>
  <summary>SUCCESSFUL RUN</summary>
  <pre>2024-10-10T07:50:46.7682441Z ##[debug]Evaluating condition for step: 'use .NET SDK 8.x'
2024-10-10T07:50:46.7684197Z ##[debug]Evaluating: SucceededNode()
2024-10-10T07:50:46.7684517Z ##[debug]Evaluating SucceededNode:
2024-10-10T07:50:46.7685665Z ##[debug]=> True
2024-10-10T07:50:46.7686022Z ##[debug]Result: True
2024-10-10T07:50:46.7686375Z ##[section]Starting: use .NET SDK 8.x
2024-10-10T07:50:46.7693389Z ==============================================================================
2024-10-10T07:50:46.7693565Z Task         : Use .NET Core
2024-10-10T07:50:46.7693661Z Description  : Acquires a specific version of the .NET Core SDK from the internet or the local cache and adds it to the PATH. Use this task to change the version of .NET Core used in subsequent tasks. Additionally provides proxy support.
2024-10-10T07:50:46.7693965Z Version      : 2.247.1
2024-10-10T07:50:46.7694052Z Author       : Microsoft Corporation
2024-10-10T07:50:46.7694168Z Help         : https://aka.ms/AA4xgy0
2024-10-10T07:50:46.7694261Z ==============================================================================
2024-10-10T07:50:46.8808913Z ##[debug]Using node path: /home/vsts/agents/3.245.0/externals/node20_1/bin/node
2024-10-10T07:50:46.9778856Z ##[debug]system.debug=True
2024-10-10T07:50:46.9781179Z ##[debug]DistributedTask.Tasks.Node.SkipDebugLogsWhenDebugModeOff=True
2024-10-10T07:50:46.9899330Z ##[debug]agent.TempDirectory=/home/vsts/work/_temp
2024-10-10T07:50:47.0637837Z ##[debug]loading inputs and endpoints
2024-10-10T07:50:47.0661845Z ##[debug]loading INPUT_PACKAGETYPE
2024-10-10T07:50:47.0687162Z ##[debug]loading INPUT_USEGLOBALJSON
2024-10-10T07:50:47.0692678Z ##[debug]loading INPUT_WORKINGDIRECTORY
2024-10-10T07:50:47.0697128Z ##[debug]loading INPUT_VERSION
2024-10-10T07:50:47.0701359Z ##[debug]loading INPUT_INCLUDEPREVIEWVERSIONS
2024-10-10T07:50:47.0707158Z ##[debug]loading INPUT_INSTALLATIONPATH
2024-10-10T07:50:47.0712900Z ##[debug]loading INPUT_PERFORMMULTILEVELLOOKUP
2024-10-10T07:50:47.0719607Z ##[debug]loading INPUT_REQUESTTIMEOUT
2024-10-10T07:50:47.0724176Z ##[debug]loading ENDPOINT_AUTH_SYSTEMVSSCONNECTION
2024-10-10T07:50:47.0731305Z ##[debug]loading ENDPOINT_AUTH_SCHEME_SYSTEMVSSCONNECTION
2024-10-10T07:50:47.0737803Z ##[debug]loading ENDPOINT_AUTH_PARAMETER_SYSTEMVSSCONNECTION_ACCESSTOKEN
2024-10-10T07:50:47.0753717Z ##[debug]loading SECRET_SYSTEM_ACCESSTOKEN
2024-10-10T07:50:47.0761177Z ##[debug]loaded 12
2024-10-10T07:50:47.0775492Z ##[debug]Agent.ProxyUrl=undefined
2024-10-10T07:50:47.0780791Z ##[debug]Agent.CAInfo=undefined
2024-10-10T07:50:47.0785761Z ##[debug]Agent.ClientCert=undefined
2024-10-10T07:50:47.0789331Z ##[debug]Agent.SkipCertValidation=undefined
2024-10-10T07:50:47.1419486Z ##[debug]Agent.ProxyUrl=undefined
2024-10-10T07:50:47.1421784Z ##[debug]Agent.CAInfo=undefined
2024-10-10T07:50:47.1423065Z ##[debug]Agent.ClientCert=undefined
2024-10-10T07:50:47.1423702Z ##[debug]check path : /home/vsts/work/_tasks/UseDotNet_b0ce7256-7898-45d3-9cb5-176b752bfea6/2.247.1/node_modules/azure-pipelines-tool-lib/lib.json
2024-10-10T07:50:47.1428026Z ##[debug]adding resource file: /home/vsts/work/_tasks/UseDotNet_b0ce7256-7898-45d3-9cb5-176b752bfea6/2.247.1/node_modules/azure-pipelines-tool-lib/lib.json
2024-10-10T07:50:47.1429039Z ##[debug]system.culture=en-US
2024-10-10T07:50:47.1443896Z ##[debug]Set default DNS lookup order to ipv4 first
2024-10-10T07:50:47.1444682Z ##[debug]Set default auto select family to false
2024-10-10T07:50:47.1600896Z ##[debug]Setting resource path to /home/vsts/work/_tasks/UseDotNet_b0ce7256-7898-45d3-9cb5-176b752bfea6/2.247.1/task.json
2024-10-10T07:50:47.1602459Z ##[debug]check path : /home/vsts/work/_tasks/UseDotNet_b0ce7256-7898-45d3-9cb5-176b752bfea6/2.247.1/task.json
2024-10-10T07:50:47.1603136Z ##[debug]adding resource file: /home/vsts/work/_tasks/UseDotNet_b0ce7256-7898-45d3-9cb5-176b752bfea6/2.247.1/task.json
2024-10-10T07:50:47.1604147Z ##[debug]system.culture=en-US
2024-10-10T07:50:47.1631083Z ##[debug]check path : /home/vsts/work/_tasks/UseDotNet_b0ce7256-7898-45d3-9cb5-176b752bfea6/2.247.1/node_modules/azure-pipelines-tasks-packaging-common/module.json
2024-10-10T07:50:47.1632254Z ##[debug]adding resource file: /home/vsts/work/_tasks/UseDotNet_b0ce7256-7898-45d3-9cb5-176b752bfea6/2.247.1/node_modules/azure-pipelines-tasks-packaging-common/module.json
2024-10-10T07:50:47.1632850Z ##[debug]system.culture=en-US
2024-10-10T07:50:47.1641314Z ##[debug]useGlobalJson=false
2024-10-10T07:50:47.1643096Z ##[debug]packageType=sdk
2024-10-10T07:50:47.1659152Z ##[debug]version=8.x
2024-10-10T07:50:47.1660012Z ##[debug]vsVersion=undefined
2024-10-10T07:50:47.1660491Z ##[debug]nugetVersion=undefined
2024-10-10T07:50:47.1660954Z ##[debug]installationPath=/opt/hostedtoolcache/dotnet
2024-10-10T07:50:47.1661686Z ##[debug]performMultiLevelLookup=false
2024-10-10T07:50:47.1662156Z ##[debug]includePreviewVersions=false
2024-10-10T07:50:47.1662614Z ##[debug]workingDirectory=/home/vsts/work/1/s
2024-10-10T07:50:47.1663052Z ##[debug]agent.proxyurl=undefined
2024-10-10T07:50:47.1663492Z ##[debug]requestTimeout=300000
2024-10-10T07:50:47.1663979Z ##[debug]AZURE_HTTP_USER_AGENT=VSTS_8fd5c2b8-726e-4003-977f-90eee064441e_build_211_0
2024-10-10T07:50:47.1673657Z Tool to install: .NET Core sdk version 8.x.
2024-10-10T07:50:47.8412645Z ##[debug]Finding channel 8.0 for version 8.x
2024-10-10T07:50:49.5148910Z Found version 8.0.403 in channel 8.0 for user specified version spec: 8.x
2024-10-10T07:50:49.5628837Z ##[debug]Processed: ##vso[telemetry.publish area=TaskDeploymentMethod;feature=UseDotNetV2]{"userVersion":"8.x", "resolvedVersion":"8.0.403"}
2024-10-10T07:50:49.5629923Z get-os-distro: Error: Distribution specific OS name and version could not be detected: UName = Linux
2024-10-10T07:50:49.5630783Z ##[debug]isExplicit: 8.0.403
2024-10-10T07:50:49.5631241Z ##[debug]explicit? true
2024-10-10T07:50:49.5631533Z Version 8.0.403 was not found in cache.
2024-10-10T07:50:49.5631792Z Getting URL to download .NET Core sdk version: 8.0.403.
2024-10-10T07:50:49.5632073Z Detecting OS platform to find correct download package for the OS.
2024-10-10T07:50:49.5632579Z ##[debug]which '/home/vsts/work/_tasks/UseDotNet_b0ce7256-7898-45d3-9cb5-176b752bfea6/2.247.1/externals/get-os-distro.sh'
2024-10-10T07:50:49.5633196Z ##[debug]found: '/home/vsts/work/_tasks/UseDotNet_b0ce7256-7898-45d3-9cb5-176b752bfea6/2.247.1/externals/get-os-distro.sh'
2024-10-10T07:50:49.5633807Z ##[debug]which '/home/vsts/work/_tasks/UseDotNet_b0ce7256-7898-45d3-9cb5-176b752bfea6/2.247.1/externals/get-os-distro.sh'
2024-10-10T07:50:49.5634412Z ##[debug]found: '/home/vsts/work/_tasks/UseDotNet_b0ce7256-7898-45d3-9cb5-176b752bfea6/2.247.1/externals/get-os-distro.sh'
2024-10-10T07:50:49.5635023Z ##[debug]exec tool: /home/vsts/work/_tasks/UseDotNet_b0ce7256-7898-45d3-9cb5-176b752bfea6/2.247.1/externals/get-os-distro.sh
2024-10-10T07:50:49.5635518Z ##[debug]arguments:
2024-10-10T07:50:49.5636054Z [command]/home/vsts/work/_tasks/UseDotNet_b0ce7256-7898-45d3-9cb5-176b752bfea6/2.247.1/externals/get-os-distro.sh
2024-10-10T07:50:49.5636369Z Primary:linux-x64
2024-10-10T07:50:49.5636576Z Legacy:-x64
2024-10-10T07:50:49.5636802Z Detected platform (Primary): linux-x64
2024-10-10T07:50:49.5637035Z Detected platform (Legacy): -x64
2024-10-10T07:50:49.5637430Z ##[debug]Got download URL for platform with rid: linux-x64
2024-10-10T07:50:49.5638293Z ##[debug]
2024-10-10T07:50:49.5638702Z ##[debug]Agent.Version=3.245.0
2024-10-10T07:50:49.5639162Z ##[debug]Agent.TempDirectory=/home/vsts/work/_temp
2024-10-10T07:50:49.5639620Z ##[debug]testing directory '/home/vsts/work/_temp'
2024-10-10T07:50:49.5640319Z Downloading: https://download.visualstudio.microsoft.com/download/pr/ca6cd525-677e-4d3a-b66c-11348a6f920a/ec395f498f89d0ca4d67d903892af82d/dotnet-sdk-8.0.403-linux-x64.tar.gz
2024-10-10T07:50:49.5640892Z ##[debug]destination /home/vsts/work/_temp/9317eaa8-319b-4cfc-987c-a71c59676f60
2024-10-10T07:50:49.5641350Z ##[debug]downloading
2024-10-10T07:50:49.6176423Z ##[debug]Content-Length of downloaded file: 212874079
2024-10-10T07:50:49.6178015Z ##[debug]creating stream
2024-10-10T07:50:49.6212474Z ##[debug]file write stream opened. fd: 19
2024-10-10T07:50:49.6213364Z ##[debug]subscribing to message read stream events...
2024-10-10T07:50:49.6217623Z ##[debug]successfully subscribed to message read stream events
2024-10-10T07:50:51.3386963Z ##[debug]download complete
2024-10-10T07:50:51.3388007Z ##[debug]Downloaded file size: 212874079 bytes
2024-10-10T07:50:51.3389493Z Extracting downloaded package /home/vsts/work/_temp/9317eaa8-319b-4cfc-987c-a71c59676f60.
2024-10-10T07:50:51.3390092Z ##[debug]Agent.TempDirectory=/home/vsts/work/_temp
2024-10-10T07:50:51.3392428Z Extracting archive
2024-10-10T07:50:51.3393680Z ##[debug]Agent.Version=3.245.0
2024-10-10T07:50:51.3394272Z ##[debug]Agent.TempDirectory=/home/vsts/work/_temp
2024-10-10T07:50:51.3395383Z ##[debug]testing directory '/home/vsts/work/_temp/4687e1a7-d795-4b34-9b34-7e221fcf56b0'
2024-10-10T07:50:51.3396330Z ##[debug]testing directory '/home/vsts/work/_temp'
2024-10-10T07:50:51.3396956Z ##[debug]mkdir '/home/vsts/work/_temp/4687e1a7-d795-4b34-9b34-7e221fcf56b0'
2024-10-10T07:50:51.3398825Z ##[debug]which 'tar'
2024-10-10T07:50:51.3401591Z ##[debug]found: '/usr/bin/tar'
2024-10-10T07:50:51.3402613Z ##[debug]/usr/bin/tar arg: ["xC","/home/vsts/work/_temp/4687e1a7-d795-4b34-9b34-7e221fcf56b0","-f","/home/vsts/work/_temp/9317eaa8-319b-4cfc-987c-a71c59676f60"]
2024-10-10T07:50:51.3405475Z ##[debug]exec tool: /usr/bin/tar
2024-10-10T07:50:51.3406073Z ##[debug]arguments:
2024-10-10T07:50:51.3406565Z ##[debug]   xC
2024-10-10T07:50:51.3407123Z ##[debug]   /home/vsts/work/_temp/4687e1a7-d795-4b34-9b34-7e221fcf56b0
2024-10-10T07:50:51.3407673Z ##[debug]   -f
2024-10-10T07:50:51.3408213Z ##[debug]   /home/vsts/work/_temp/9317eaa8-319b-4cfc-987c-a71c59676f60
2024-10-10T07:50:51.3409049Z [command]/usr/bin/tar xC /home/vsts/work/_temp/4687e1a7-d795-4b34-9b34-7e221fcf56b0 -f /home/vsts/work/_temp/9317eaa8-319b-4cfc-987c-a71c59676f60
2024-10-10T07:50:51.5615052Z ##[debug]Agent environment resources - Disk: / Available 20328.57 MB out of 74244.74 MB, Memory: Used 980.00 MB out of 6921.00 MB, CPU: Usage 0.50%
2024-10-10T07:50:56.5626204Z ##[debug]Agent environment resources - Disk: / Available 19833.64 MB out of 74244.74 MB, Memory: Used 1019.00 MB out of 6921.00 MB, CPU: Usage 68.33%
2024-10-10T07:50:57.5334823Z 
2024-10-10T07:50:57.5367653Z ##[debug]Process exited with code 0 and signal null for tool '/usr/bin/tar'
2024-10-10T07:50:57.5371318Z ##[debug]STDIO streams have closed and received exit code 0 and signal null for tool '/usr/bin/tar'
2024-10-10T07:50:57.5374387Z ##[debug]Copying all root folders into installation path: /opt/hostedtoolcache/dotnet
2024-10-10T07:50:58.6877719Z ##[debug]Comparing if version being installed 8.0.403 is greater than already installed version with folder name 6.0.427
2024-10-10T07:50:58.6878895Z ##[debug]isExplicit: 6.0.427
2024-10-10T07:50:58.6880372Z ##[debug]explicit? true
2024-10-10T07:50:58.6880976Z ##[debug]isExplicit: 8.0.403
2024-10-10T07:50:58.6881634Z ##[debug]explicit? true
2024-10-10T07:50:58.6882469Z ##[debug]Comparing if version being installed 8.0.403 is greater than already installed version with folder name 8.0.403
2024-10-10T07:50:58.6884215Z ##[debug]isExplicit: 8.0.403
2024-10-10T07:50:58.6884856Z ##[debug]explicit? true
2024-10-10T07:50:58.6886077Z ##[debug]isExplicit: 8.0.403
2024-10-10T07:50:58.6886673Z ##[debug]explicit? true
2024-10-10T07:50:58.6887499Z ##[debug]Comparing if version being installed 8.0.403 is greater than already installed version with version complete file name 6.0.427
2024-10-10T07:50:58.6889056Z ##[debug]isExplicit: 6.0.427
2024-10-10T07:50:58.6889548Z ##[debug]explicit? true
2024-10-10T07:50:58.6889995Z ##[debug]isExplicit: 8.0.403
2024-10-10T07:50:58.6890412Z ##[debug]explicit? true
2024-10-10T07:50:58.6890924Z ##[debug]Version: 8.0.403 is the latest among the versions present at path: /opt/hostedtoolcache/dotnet
2024-10-10T07:50:58.6891502Z ##[debug]Copying root files (such as dotnet.exe) into installation path: /opt/hostedtoolcache/dotnet
2024-10-10T07:50:58.9137261Z ##[debug]Creating installation complete marker file for .Net core 8.0.403 version sdk
2024-10-10T07:50:58.9141222Z Successfully installed .NET Core sdk version 8.0.403.
2024-10-10T07:50:58.9157568Z ##[debug]Agent.Version=3.245.0
2024-10-10T07:50:58.9158652Z ##[debug]Processed: ##vso[task.prependpath]/opt/hostedtoolcache/dotnet
2024-10-10T07:50:58.9159149Z ##[debug]set DOTNET_ROOT=/opt/hostedtoolcache/dotnet
2024-10-10T07:50:58.9163882Z ##[debug]Processed: ##vso[task.setvariable variable=DOTNET_ROOT;isOutput=false;issecret=false;]/opt/hostedtoolcache/dotnet
2024-10-10T07:50:58.9164556Z ##[debug]set DOTNET_MULTILEVEL_LOOKUP=0
2024-10-10T07:50:58.9165451Z ##[debug]Processed: ##vso[task.setvariable variable=DOTNET_MULTILEVEL_LOOKUP;isOutput=false;issecret=false;]0
2024-10-10T07:50:58.9166961Z Creating global tool path and pre-pending to PATH.
2024-10-10T07:50:58.9167499Z ##[debug]testing directory '/home/vsts/.dotnet/tools'
2024-10-10T07:50:58.9167973Z ##[debug]Agent.Version=3.245.0
2024-10-10T07:50:58.9168558Z ##[debug]Processed: ##vso[task.prependpath]/home/vsts/.dotnet/tools
2024-10-10T07:50:58.9169039Z ##[debug]Agent.ProxyUrl=undefined
2024-10-10T07:50:58.9169794Z ##[debug]task result: Succeeded
2024-10-10T07:50:58.9172033Z ##[debug]Processed: ##vso[task.complete result=Succeeded;]
2024-10-10T07:50:58.9288222Z ##[section]Finishing: use .NET SDK 8.x
 </pre>
</details>

### Repro steps

steps:
- task: UseDotNet@2
            displayName: 'use .NET SDK 8.x'
            inputs:
              packageType: 'sdk'
              version: '8.x'

**Comment by sanderaernouts** [comment]
It looks a lot like a retry or time(out) related issue to me. I checked the code of this task but I cannot find any (obvious) code path that would allow the download request to timeout without an error being thrown. But commit https://github.com/microsoft/azure-pipelines-tasks/commit/435884d969a0d211cb3654fb0ba56bfc68cb0605 added a timeout to the request options and I suspect something is wrong with the retry/timeout logic.

**Comment by sanderaernouts** [comment]
I tried to reproduce the issue and failed. Checked the last pipeline that failed again and I now see that was using `2.246.1` instead of `2.247.1`. With `2.246.1` I can reproduce the issue quite reliably. So it looks like this issue was resolved in `2.247.1`
